bugbot/rules/file_crash_bug.py

```diff
@@ -0,0 +1,156 @@
+
+
+class FileCrashBug(BzCleaner):
+    """File bugs for new actionable crashes."""
+
+    # NOTE: If you make changes that affect the output of the rule, you should
+    # increment this number. This is needed in the experimental phase only.
+    VERSION = 1
+    MAX_BUG_TITLE_LENGTH = 255
+
+    def __init__(self):
+        super().__init__()
+
+        self.bug_description_template = jinja2.Environment(
+            loader=jinja2.FileSystemLoader("templates")
+        ).get_template("file_crash_bug_description.md.jinja")
+
+    def description(self):
+        return "New actionable crashes"
+
+    def columns(self):
+        return ["component", "id", "summary"]
+
+    def get_bz_params(self, date):
+        return {
+            "resolution": ["---", "FIXED"],
+            "keywords": ["feature", "regression"],
+            "keywords_type": "allwords",
+        }
+
+    def get_bugs(self, date):
+        self.query_url = None
+        bugs = {}
+
+        signatures = SignaturesDataFetcher.find_new_actionable_crashes(
+            "Firefox", "nightly"
+        )
+
+        for signature in signatures.analyze():
+            logger.debug("Generating bug for signature: %s", signature.signature_term)
+
+            title = f"Crash in [@ {signature.signature_term}]"
+            if len(title) > self.MAX_BUG_TITLE_LENGTH:
+                title = title[: self.MAX_BUG_TITLE_LENGTH - 3] + "..."
+
+            # TODO: Handle cases where the regressor is a security bug. I such
+            # cases, we may want to fil the bug as security bug.
+
+            flags = None
+            if signature.regressed_by:
+                # TODO: check user activity and if the ni? is open
+                flags = [
+                    {
+                        "name": "needinfo",
+                        "requestee": signature.regressed_by_author["name"],
+                        "status": "?",
+                        "new": "true",
+                    }
+                ]
+
+            report = signature.fetch_representing_processed_crash()
+            description = self.bug_description_template.render(
+                {
+                    **socorro_util.generate_bug_description_data(report),
+                    "signature": signature,
+                    "needinfo_regression_author": bool(flags),
+                }
+            )
+
+            # TODO: Provide the following information:
+            # [X] Crash signature
+            # [X] Top 10 frames of crashing thread
+            # [X] Component
+            # [X] The kind of crash
+            # [ ] Regression window
+            # [X] Inducing patch
+            # [X] Reason
+            # [X] Regressed by
+            # [X] Platform
+            # [ ] Firefox status flags
+            # [ ] Severity
+            # [ ] Time correlation
+            # [X] User comments
+            # [ ] Crash address commonalities
+            # [ ] Estimated future crash volume
```

Comment:
Is this all you had in the planning doc?

Reply:
Yes, in the same order.